### PR TITLE
[feat] Print test dependencies in failure report

### LINKED ISSUE #1414: Print dependencies of tests in failure report
@rsarm I think you should also store this information in your #1377 PR.

### PR DESCRIPTION
**Pull request by ekouts**
Closes #1414.

Example of output:
```
$ ./bin/reframe -c unittests/resources/checks_unlisted/deps_complex.py -r
...
==============================================================================
SUMMARY OF FAILURES
...
------------------------------------------------------------------------------
FAILURE INFO for T9
  * Test Description: T9
  * System partition: dom:mc
  * Environment: PrgEnv-intel
  * Stage directory: None
  * Node list: None
  * Job type: batch job (id=None)
  * Dependencies (conceptual): ['T8']
  * Dependencies (actual): [('T8', 'dom:mc', 'PrgEnv-intel')]
  * Maintainers: []
  * Failing phase: startup
  * Rerun with '-n T9 -p PrgEnv-intel --system dom:mc'
  * Reason: task dependency error: dependencies failed
------------------------------------------------------------------------------
FAILURE INFO for T9
  * Test Description: T9
  * System partition: dom:mc
  * Environment: PrgEnv-pgi
  * Stage directory: None
  * Node list: None
  * Job type: batch job (id=None)
  * Dependencies (conceptual): ['T8']
  * Dependencies (actual): [('T8', 'dom:mc', 'PrgEnv-pgi')]
  * Maintainers: []
  * Failing phase: startup
  * Rerun with '-n T9 -p PrgEnv-pgi --system dom:mc'
  * Reason: task dependency error: dependencies failed
...
```

### CODE REVIEW

**Review by vkarak — CHANGES_REQUESTED** [review]
You should also bump the data version of the report here:

https://github.com/eth-cscs/reframe/blob/044397958f6cc6eba0a5272cd26dbbb03c0959d8/reframe/frontend/cli.py#L597

The idea is to bump the data format minor version every time we do an incremental change and the major version every time there is a breaking change.

**Review by rsarm — APPROVED** [review]

File: reframe/frontend/statistics.py [hunk 1]
```diff
@@ -82,7 +82,11 @@ def json(self, force=False):
                 entry = {
                     'build_stderr': None,
                     'build_stdout': None,
-                    'dependencies': [
+                    'dependencies_actual': [
+                        (d.check.name, d.partition.fullname, d.environ.name)
```
Comment (vkarak) [inline reframe/frontend/statistics.py:86]: If I'm not mistaken you can write this simply as `(*d)`.

**Review by vkarak — APPROVED** [review]
Lgtm, I only have a minor comment.

File: reframe/frontend/statistics.py (hunk 1 above)
Comment (vkarak) [reply reframe/frontend/statistics.py:86]: No, ignore this. The correct is `[*d]`, but it wouldn't work, because it returns the objects not the names.

**Pull request merged**